src/Symfony/Component/HttpKernel/Controller/ControllerResolver.php

```diff
@@ -47,23 +43,42 @@ public function getController(Request $request)
         }
 
         if (is_array($controller)) {
+            if (isset($controller[0]) && is_string($controller[0]) && isset($controller[1])) {
+                try {
+                    $controller[0] = $this->instantiateController($controller[0]);
+                } catch (\Error | \LogicException $e) {
+                    try {
+                        // We cannot just check is_callable but have to use reflection because a non-static method
+                        // can still be called statically in PHP but we don't want that. This is deprecated in PHP 7, so we
+                        // could simplify this with PHP 8.
+                        if ((new \ReflectionMethod($controller[0], $controller[1]))->isStatic()) {
+                            return $controller;
+                        }
+                    } catch (\ReflectionException $reflectionException) {
+                        throw $e;
+                    }
+
+                    throw $e;
+                }
+            }
+
+            if (!is_callable($controller)) {
```

Comment:
What about using http://php.net/manual/en/closure.fromcallable.php to ensure it's callable?

Reply:
I don't know what you intend. In which way does Closure::fromCallable help here?

Reply:
The `Closure::fromCallable` method calls `is_callable` for you.
Then, It "standardize" how we (will) manage callable

Reply:
Using a closure here provides no benefit and the error it would return for wrong callables is worse than what `getControllerError` does. For example it does not hint that an __invoke method might be missing. https://3v4l.org/fJJNs